### PR TITLE
Add dataset has, size and matches matchers

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Adds two matchers to improve https://github.com/libero/article-store/blob/ffd5ed504c54081997658193fac8fd78e1009f2f/test/routes/article-list.test.ts#L29-L33:

- `toHaveQuad` to check that `DatasetCore.has()` is `true`
- `toHaveQuads` to check `DatasetCore.size`

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
Agree the naming's not ideal.

> Both names may be a bit too generic, and may clash with similar matchers in the future (such as membership checks for `Store`).

Did wonder about adding support for both `DatasetCore` _and_ `Source`/`Stream`, rather than having separate matchers. (Seems like checking that it's both a dataset and something else is two separate matchers, and increases the wordiness of the names.)

**Comment by rubensworks** [comment]
> Did wonder about adding support for both DatasetCore and Source/Stream

I would prefer not to confuse those operations, as they may not always be semantically equivalent.
While for DatasetCore and Source, repeated membership operations are possible,
this is not the case for Stream, as the matcher would have to consume the whole stream to perform this membership check, which makes the stream unusable afterwards.

**Comment by thewilkybarkid** [comment]
Added `toBeRdfDatasetMatching` to look at `DatasetCore.match` directly.

Will look at the missing tests now.

File: README.md [hunk 1]
```diff
@@ -104,6 +104,39 @@ Terms are compared under the semantics of _toEqualRdfTerm_.
...
 
+#### toBeRdfDatasetContaining
+
+Check if a dataset contains one or more quads.
+
+```js
+const q1 = quad(namedNode('s1'), namedNode('p1'), namedNode('o1'), namedNode('g1'));
+const q2 = quad(namedNode('s2'), namedNode('p2'), namedNode('o2'), namedNode('g2'));
+const d = dataset([q1, q2]);
+expect(d).toBeRdfDatasetContaining(q1, q2);
```
Comment (rubensworks) [inline README.md]: Perhaps you can also add `expect(d).toBeRdfDatasetContaining(q1);` here as an example, to show that any number of quads can be entered.

File: README.md [hunk 2]
```diff
@@ -104,6 +104,39 @@ Terms are compared under the semantics of _toEqualRdfTerm_.
 expect([namedNode('t1'), namedNode('t2')]).toEqualRdfTermArray([namedNode('t2'), namedNode('t3')]);
 ```
 
+#### toBeRdfDatasetContaining
+
+Check if a dataset contains one or more quads.
```
Comment (rubensworks) [inline README.md]: Could you mention here that any number of quads can be passed, and that *all* of these quads must be present in the dataset before the test passes? (as opposed to just one of the given quads)

File: README.md [hunk 3]
```diff
@@ -104,6 +104,39 @@ Terms are compared under the semantics of _toEqualRdfTerm_.
...
+```js
+const q1 = quad(namedNode('s1'), namedNode('p1'), namedNode('o1'), namedNode('g1'));
+const q2 = quad(namedNode('s2'), namedNode('p2'), namedNode('o2'), namedNode('g2'));
+const d = dataset([q1, q2]);
+expect(d).toBeRdfDatasetContaining(q1, q2);
+```
+
+#### toBeRdfDatasetMatching
+
+Check if a dataset contains a certain number of matching quads (default 1).
```
Comment (rubensworks) [inline README.md]:
```suggestion
Check if a dataset contains *exactly* the given amount of matching quads (default 1).
```

Comment (rubensworks) [reply README.md]: To clarify that it is an `==`, not a `>=`.

**Comment by thewilkybarkid** [comment]
Hopefully all good now!

**Comment by rubensworks** [comment]
Looks perfect, thanks!
Released as 1.4.0.

**Pull request merged**